Public/Src/Engine/Processes/ExternalVMSandboxedProcess.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using BuildXL.Native.IO;
+using BuildXL.Utilities;
+using BuildXL.Utilities.VmCommandProxy;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// Sandboxed process that will be executed in VM.
+    /// </summary>
+    public class ExternalVmSandboxedProcess : ExternalSandboxedProcess
+    {
+        private int m_processId = -1;
+
+        /// <inheritdoc />
+        public override int ProcessId => m_processId != -1 ? m_processId : (m_processId = Process?.Id ?? -1);

```

Comment:
I think this should be equivalent to `m_processExecutor?.ProcessId ?? -1`
```suggestion
        public override int ProcessId => m_processExecutor?.ProcessId ?? -1;
```